src/Acuminator/Acuminator.Analyzers/StaticAnalysis/DacFieldAndReferencedFieldMismatch/DacFieldAndReferencedFieldMismatchAnalyzer.cs

```diff
@@ -0,0 +1,220 @@
+				Descriptors.PX1078_DacFieldHasDataTypeAttributesIncompatibleWithReferencedFieldsCLRType);
+
+		public override void Analyze(SymbolAnalysisContext context, PXContext pxContext, DacSemanticModel dac)
+		{
+			context.CancellationToken.ThrowIfCancellationRequested();
+
+			var propertiesToCheck = dac.DeclaredDacFieldPropertiesWithAcumaticaAttributes
+				.Select(p => (p, TryGetForeignReferenceAttribute(pxContext, p)))
+				.Where(p => p.Item2 is not null);
+
+			foreach (var (property, foreignReferenceAttribute) in propertiesToCheck)
+			{
+				context.CancellationToken.ThrowIfCancellationRequested();
+
+				CheckProperty(context, property, foreignReferenceAttribute!, dac, pxContext);
+			}
+		}
+
+		private static DacFieldAttributeInfo? TryGetForeignReferenceAttribute(PXContext pxContext, DacPropertyInfo property) =>
+			property.Attributes.FirstOrDefault(x => ContainsAttributeMatching(x, pxContext.AttributeTypes.PXSelectorAttribute.Type));
+
+		private void CheckProperty(SymbolAnalysisContext context, DacPropertyInfo property, DacFieldAttributeInfo foreignReferenceAttribute,
+			DacSemanticModel dac, PXContext pxContext)
+		{
+			var foreignField = ExtractForeignFieldSymbol(foreignReferenceAttribute, pxContext);
+
+			if (foreignField is null)
+				return;
+
+			var parentType = (foreignField.ContainingSymbol as INamedTypeSymbol)!;
+
+			var foreignDac = GetOrInferModel(parentType, pxContext, cancellation: context.CancellationToken);
+
+			if (foreignDac?.PropertiesByNames.TryGetValue(foreignField.Name, out var foreignProperty) != true)
+				return;
+
+			var foreignCLRType = foreignProperty!.PropertyType;
+
+			if (HaveDifferentUnderlyingType(property, foreignCLRType))
+				ReportTypeMismatch(context, pxContext, property, foreignProperty);
+			else
+			{
+				var localTypeAttributes = property.Attributes.Where(
+					x => ContainsAttributeMatching(x, pxContext.FieldAttributes.PXDBFieldAttribute)
+					|| ContainsAttributeMatchingOneOf(x, pxContext.FieldAttributes.UnboundDataTypeAttributes))
+					.ToList(property.Attributes.Length);
+
+				if (localTypeAttributes.Count == 0)
+					return;
+
+				var incompatibleAttribute = localTypeAttributes
+					.FirstOrDefault(attr => IsAttributeIncompatibleWithCLRType(attr, foreignCLRType, pxContext));
+
+				if (incompatibleAttribute is not null)
+					ReportTypeIncompatibility(context, pxContext, incompatibleAttribute, property, foreignProperty);
+
+				var foreignTypeAttribute = foreignProperty!.Attributes.FirstOrDefault(
+					x => ContainsAttributeMatching(x, pxContext.FieldAttributes.PXDBFieldAttribute)
+					|| ContainsAttributeMatchingOneOf(x, pxContext.FieldAttributes.UnboundDataTypeAttributes));
+
+				// agreed to stop checking property here as the code may be incomplete at this point
+				if (foreignTypeAttribute is null)
+					return;
+
+				var localTypeAttribute = localTypeAttributes.FirstOrDefault(x => HaveDifferentFieldSize(x, foreignTypeAttribute, pxContext));
```

Comment:
@SENya1990, if it's part of coding guidelines, then ok, but I doubt your suggestion about naming really makes code more readable. Regarding size calculation - ok

Reply:
I disagree about the naming being less readable with more context in the name. However, since this is a local variable which is used only once near its declaration, I won't insist on this remark, it can be omitted.